include/aspect/simulator.h

```diff
@@ -1236,6 +1260,13 @@ namespace aspect
        */
       std_cxx11::shared_ptr<MeltHandler<dim> > melt_handler;
 
+      /**
+       * Unique pointer for an instance of the NewtonHandler. This way,
+       * if we do not need the machinery for doing Newton stuff, we do
+       * not even allocate it.
+       */
+      std_cxx11::unique_ptr<NewtonHandler<dim> > newton_handler;
```

Comment:
Yes, that is the reason. So should I change the other one into a unique pointer or this one in a shared pointer, or just leave it as it is?